mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/AttachmentId.java

```diff
@@ -18,6 +18,55 @@
  ****************************************************************/
 package org.apache.james.mailbox.store.mail.model;
 
-public interface AttachmentId {
-    String serialize();
+
+import org.apache.commons.codec.digest.DigestUtils;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+public class AttachmentId {
+
+    public static AttachmentId forPayload(byte[] payload) {
+        Preconditions.checkArgument(payload != null);
```

Comment:
I prefer an `IllegalArgumentException` which means that the argument is bad than a `NullPointerException` that means nothing...